pygeoapi_iwls/iwls_api_connector.py

```diff
@@ -0,0 +1,202 @@
+
+        #  Convert datetime object back to ISO 8601 format strings
+        time_ranges_strings = [[datetime.datetime.strftime(i,'%Y-%m-%dT%H:%M:%SZ',) for i in x] for x in time_ranges]
+
+        # Get metadata
+        metadata = self._get_station_metadata(station_code)
+
+        # Get the station id from the station code
+        station_id = self._id_from_station_code(station_code)
+
+        # Use the station id in the url for the query
+        url = f'https://api-iwls.dfo-mpo.gc.ca/api/v1/stations/{station_id}/data'
+
+        return time_ranges_strings, metadata, url
+
+    def _get_timeseries(self,url: str, time_ranges_strings: pd.core.frame.DataFrame, series_code: str):
+        """
+        Send a series of queries to the IWLS API and return
+
+        :param url: url used for queries (String)
+        :param time_ranges_string: pairs of start times and end times used for queries (pd.Dataframe)
+        :param series_code: three letter identifer for time series (String)
+                            'wlo' = Observed Water Levels
+                            'wlp' = Tidal Predictions
+                            'wlf' = Quality Control Forecast
+                            'wlf-spine' = SPINE Forcast
+                            'wcs1' = Observed Surface Currents Speed
+                            'wcd1' = Observed Surface Currents Direction
+        returns: series of pairs of time stamps and water level values (json)
+        """
+        series_data = pd.DataFrame()
+        for i in time_ranges_strings:
+            params = {
+                'time-series-code':series_code,
+                'from':i[0],
+                'to': i[1]
+                }
+            r = requests.get(url=url, params=params)
+            r.raise_for_status()
+            series_data = pd.concat([series_data, pd.DataFrame.from_dict(r.json())])
+
+        if series_data.empty:
+                return json.dumps({'value':{}})
+        else:
+                series_data['eventDate'] = pd.to_datetime(series_data['eventDate'])
+                series_data = series_data.set_index('eventDate').sort_index()
+                series_data = series_data[['value']]
+
+                return series_data.to_json(date_format='iso')
+
+
+    def _get_timeseries_by_boundary(self, start_time: str, end_time: str, bbox: list,
+                                    limit: int, start_index: int):
+        """
+        Contains all logic common to WaterLevels/SurfaceCurrents IWLSConnector class to retrieve timeseries data inside of a bounding box.
+
+        :param  start_time: Start time, ISO 8601 format UTC (e.g.: 2019-11-13T19:18:00Z) (string)
+        :param  start_time: Start time, ISO 8601 format UTC (e.g.: 2019-11-13T19:18:00Z) (string)
+        :param bbox: bounding box [minx,miny,maxx,maxy] (list)
+        :param limit: number of records to return (default 10) (int)
+        :param start_index: starting record to return (default 0) (int)
```

Comment:
Default values  for start_index and limit are not being passed by the base class (they are still present in the child classes). Should be added or removed from the doc string descriptions)

Reply:
Sorry, why should they be removed? I'm not quite clear on what you mean here. Are you referring to the `default 10` and `default 0` text? I can remove that but they are still params so I think there should be something here for them.